src/lib/storage/fsst_segment/fsst_encoder.hpp

```diff
@@ -0,0 +1,192 @@
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <limits>
+#include <memory>
+#include <string>
+
+#include "storage/base_segment_encoder.hpp"
+#include "storage/value_segment.hpp"
+#include "storage/value_segment/value_segment_iterable.hpp"
+#include "types.hpp"
+#include "utils/assert.hpp"
+#include "utils/enum_constant.hpp"
+
+namespace opossum {
+
+class FSSTEncoder : public SegmentEncoder<FSSTEncoder> {
+ public:
+  static constexpr auto _encoding_type = enum_c<EncodingType, EncodingType::FSST>;
```

Comment:
Is there a reason why you are naming your public class variables and methods using an "_" prefix although they are private?

Reply:
This seems to be convention for all SegmentEncoders: 

DictionaryEncoder:
https://github.com/hyrise/hyrise/blob/d50c5db23f1f53dfa5745706ea95cbc3b4f7a0a3/src/lib/storage/dictionary_segment/dictionary_encoder.hpp#L26-L29

LZ4Segment: 
https://github.com/hyrise/hyrise/blob/d50c5db23f1f53dfa5745706ea95cbc3b4f7a0a3/src/lib/storage/lz4_segment/lz4_encoder.hpp#L46-L49